desidlas/datasets/DesiMock.py

```diff
@@ -0,0 +1,155 @@
+
+        test = np.array([True if item in dlas_data['TARGETID'] else False for item in spec[1].data['TARGETID'].copy()])
+        for item in spec[1].data['TARGETID'].copy()[~test]:
+            spec_dlas[item] = []
+
+        # read data from the fits file above, one can directly get those varibles meanings by their names.
+        spec_id = spec[1].data['TARGETID'].copy()
+        flux_b = spec[3].data.copy()
+        flux_r = spec[8].data.copy()
+        flux_z = spec[13].data.copy()
+        flux = np.hstack((flux_b,flux_r,flux_z))
+        ivar_b = spec[4].data.copy()
+        ivar_r = spec[9].data.copy()
+        ivar_z = spec[14].data.copy()
+        error = 1./np.sqrt(np.hstack((ivar_b,ivar_r,ivar_z)))
+        self.split_point_br = flux_b.shape[1]
+        self.split_point_rz = flux_b.shape[1]+flux_r.shape[1]
+        z_qso = zbest[1].data['Z'].copy()
+        ra = spec[1].data['TARGET_RA'].copy()
+        dec = spec[1].data['TARGET_DEC'].copy()
+
+        self.data = {spec_id[i]:{'FLUX':flux[i],'ERROR': error[i], 'z_qso':z_qso[i] , 'RA': ra[i], 'DEC':dec[i], 'DLAS':spec_dlas[spec_id[i]]} for i in range(len(spec_id))}
+
+    def get_sightline(self, id, camera = 'all', rebin=False, normalize=False):
+        """
+        using id(int) as index to retrive each spectra in DesiMock's dataset, return  a Sightline object.
+        ---------------------------------------------------------------------------------------------------
+        parameters:
+        id: spectra's id , a unique number for each spectra.
+        camera: str, 'b' : Load up the wavelength and data for the blue camera., 'r': Load up the wavelength and data for the r camera,
+                     'z' : Load up the wavelength and data for the z camera, 'all':  Load up the wavelength and data for all cameras.
+        rebin: bool, if True rebin the spectra to the best dlambda/lambda, default False,
+        normalize: bool, if True normalize the spectra, using the slice of flux from wavelength ~1070 to 1170, default False.
+        ---------------------------------------------------------------------------------------------------
+        return:
+        sightline: dla_cnn.data_model.Sightline.Sightline object
+        """
+        assert camera in ['all', 'r', 'z', 'b'], "No such camera! The parameter 'camera' must be in ['all', 'r', 'b', 'z']"
+        sightline = Sightline(id)
+
+        # this inside method can get the data(wavelength, flux, error) from the start_point(int) to end_point(int)
+        def get_data(start_point=0, end_point=self.data_size):
+            """
+            
+            Parameters
+            ----------
+            start_point: int, the start index of the slice of the data(wavelength, flux, error), default 0
+            end_point: int, the end index of the slice of the data(wavelength, flux, error), default the length of the data array
+            
+            Returns
+            -------
+            
```

Comment:
Describe the `sightline` that is returned

Reply:
Done